lib/pg_query/parse.rb

```diff
@@ -122,6 +122,18 @@ def load_objects! # rubocop:disable Metrics/CyclomaticComplexity
             when :SETOP_NONE
               (statement.select_stmt.from_clause || []).each do |item|
                 from_clause_items << { item: item, type: :select }
+
+                join_expr_quals = []
+                statement.select_stmt.from_clause.each do |from_clause|
+                  next unless from_clause.join_expr
+                  curr_join_expr = from_clause.join_expr
+                  while curr_join_expr
+                    join_expr_quals << curr_join_expr.quals
+                    curr_join_expr = curr_join_expr.larg.join_expr
+                  end
+                end
+
+                subselect_items.concat(join_expr_quals)
```

Comment:
This seems to work, but I'd appreciate someone more familiar with the code taking a look. It's possible I may be overlooking something.